mirrord/agent/src/steal/connection.rs

```diff
@@ -147,10 +149,108 @@ impl Client {
         }
 
         let framed = HTTP_FRAMED_VERSION.matches(&self.protocol_version);
+        let chunked = HTTP_CHUNKED_VERSION.matches(&self.protocol_version);
         let tx = self.tx.clone();
 
         tokio::spawn(async move {
-            if framed {
+            // Chunked data is preferred over framed data
+            if chunked {
+                // Send headers
+                tracing::error!("starting bg task");
+                let connection_id = request.connection_id;
+                let request_id = request.request_id;
+                let (
+                    Parts {
+                        method,
+                        uri,
+                        version,
+                        headers,
+                        ..
+                    },
+                    mut body,
+                ) = request.request.into_parts();
+                match body.next_frames(true).await {
+                    Err(..) => return,
+                    Ok(Frames { frames, is_last }) => {
+                        let frames = frames
+                            .into_iter()
+                            .filter_map(|frame| {
+                                if frame.is_data() {
+                                    frame
+                                        .into_data()
+                                        .ok()
+                                        .map(|x| ChunkedFrame::Data(x.to_vec()))
+                                } else if frame.is_trailers() {
+                                    frame.into_trailers().ok().map(ChunkedFrame::Trailer)
+                                } else {
+                                    None
+                                }
```

Comment:
Since we're going to do conversion between `ChunkedFrame` and `Frame<Bytes>` in multiple places, you can implement some traits ^^
```rust
impl From<ChunkedFrame> for Frame<Bytes> {
    ...
}

impl TryFrom<Frame<Bytes>> for ChunkedFrame {
    type Error = Frame<Bytes>;

    ...
}
```
Then it's just `let frames = frames.into_iter().map(ChunkedFrame::try_from).filter_map(Result::ok).collect()`